cmake/developer_package/version.cmake

```diff
@@ -160,13 +160,17 @@ macro (ov_add_version_defines FILE TARGET)
     if(NOT EXISTS ${__version_file})
         message(FATAL_ERROR "${FILE} does not exists in current source directory")
     endif()
+    if (NOT TARGET ${TARGET})
+        message(FATAL_ERROR "Invalid target ${TARGET}")
+    endif()
+    get_target_property(__target_type ${TARGET} TYPE)
+    if (NOT __target_type MATCHES "^(.*)_LIBRARY$")
+        message(FATAL_ERROR "Target ${TARGET} is not a library")
+    endif()
+    set(__lib_type ${CMAKE_MATCH_1})
     _remove_source_from_target(${TARGET} ${FILE})
     _remove_source_from_target(${TARGET} ${__version_file})
-    if (BUILD_SHARED_LIBS)
-        add_library(${TARGET}_version OBJECT ${__version_file})
-    else()
-        add_library(${TARGET}_version STATIC ${__version_file})
-    endif()
+    add_library(${TARGET}_version ${__lib_type} ${__version_file})
```

Comment:
what if `__lib_type` is `SHARED`? I suppose we don't need shared library with version :)

Reply:
I guess I can add exception for this, though not sure the original code was resilient to this.
It appears we use MODULE so at least MODULE/OBJECT/STATIC need to be accepted.

Reply:
On the other hand not sure we need to block SHARED as it should work in principle.
Cmake allows library with no explicit type build to STATIC or SHARED based on BUILD_SHARED_LIBS.
Not sure we want to make it break in that flow even if we don't rely on that.

Reply:
version library is an internal artifact. If you create shared library with it => you expose it to end users.
We must create only static / object libs here, all other types are not accepted

Reply:
Updated